src/Implementation/ColdStorage/ColdStorage/ColdStorageProcessor.cs

```diff
@@ -0,0 +1,327 @@
+
+        public ColdStorageProcessor(
+            Func<string, IBlobWriter> blobWriterFactory,
+            CancellationToken token,
+            int warningLevel,
+            int tripLevel,
+            TimeSpan stallInterval,
+            TimeSpan logCooldownInterval,
+            string eventHubName,
+            int maxBlocks = MaxBlocks,
+            int maxBlockSize = MaxBlockSize)
+        {
+            Guard.ArgumentNotNull(blobWriterFactory, "blobWriterFactory");
+
+            _blobWriterFactory = blobWriterFactory;
+            _token = token;
+
+            _warningLevel = warningLevel;
+            _tripLevel = tripLevel;
+            _stallInterval = stallInterval;
+            _logCooldownInterval = logCooldownInterval;
+            _eventHubName = eventHubName;
+
+            _maxBlockSize = maxBlockSize;
+            _buffers = BufferManager.CreateBufferManager(maxBlocks, _maxBlockSize);
+
+            _eventHubBufferDataList = new List<BufferedFrameData>();
+        }
+
+        public Task OpenAsync(PartitionContext context)
+        {
+            ColdStorageEventSource.Log.LeaseObtained(ProcessorName, _eventHubName, context.Lease.PartitionId);
+
+            // Generate a blob writer for this partition context
+            _blobWriter = _blobWriterFactory(context.Lease.PartitionId);
+            _currentFrame = _buffers.TakeBuffer(_maxBlockSize);
+            _currentFrameLength = 0;
+            _lastEventData = null;
+            _nextWarningLogTime = null;
+
+            return Task.FromResult(false);
+        }
+
+        public async Task CloseAsync(PartitionContext context, CloseReason reason)
+        {
+            // If we have partial blocks to flush, write them to our internal buffer
+            if (_currentFrameLength > 0)
+            {
+                AddCurrentFrameToBuffer();
+            }
+
+            // If we are doing a clean shutdown (as opposed to losing the lease) flush 
+            // our in-memory data to blob storage
+            if (reason == CloseReason.Shutdown)
+            {
+                ColdStorageEventSource.Log.ShutDownInitiated(ProcessorName, _eventHubName, context.Lease.PartitionId);
+                if (_eventHubBufferDataList.Count > 0)
+                {
+                    await FlushAndCheckPointAsync(context).ConfigureAwait(false);
+                }
+            }
+            // If we just lost the lease, flush our data (the responsibility of writing this
+            // partition's data has been assigned to another instance)
+            else if (reason == CloseReason.LeaseLost)
+            {
+                ColdStorageEventSource.Log.LeaseLost(ProcessorName, _eventHubName, context.Lease.PartitionId);
+
+                ClearBufferDataList();
+            }
+        }
+
+        public Task ProcessEventsAsync(PartitionContext context, IEnumerable<EventData> events)
+        {
+            return ProcessEventsAsync(context, events, CancellationToken.None);
```

Comment:
If the event processor isn't passing in the cancellation token, then let's remove this overload.
It was added in anticipation of the event processor providing it.